src/test/java/org/jpeek/MetricsTest.java

```diff
@@ -55,229 +55,216 @@
-            new Object[] {"TwoCommonAttributes", "CCM", Double.NaN},
-            new Object[] {"TwoCommonMethods", "CCM", 0.0238d},
-            new Object[] {"Bar", "OCC", 0.75d},
-            new Object[] {"BridgeMethod", "OCC", 0.0d},
-            new Object[] {"ClassWithPublicField", "OCC", 0.5d},
-            new Object[] {
-                "IndirectlyRelatedPairs",
-                "OCC",
-                0.6666666666666666d,
-            },
-            new Object[] {"MethodMethodCalls", "OCC", 0.2d},
-            new Object[] {
-                "MethodsWithDiffParamTypes",
-                "OCC",
-                0.3333333333333333d,
-            },
-            new Object[] {"NoMethods", "OCC", 0.0d},
-            new Object[] {"OneMethodCreatesLambda", "OCC", 0.0d},
-            new Object[] {"OneVoidMethodWithoutParams", "OCC", 0.0d},
-            new Object[] {"OnlyOneMethodWithParams", "OCC", 1d},
-            new Object[] {"OverloadMethods", "OCC", 0.75d},
-            new Object[] {"TwoCommonAttributes", "OCC", 0.0d},
-            new Object[] {"TwoCommonMethods", "OCC", 0.0d},
-            new Object[] {"WithoutAttributes", "OCC", 0.0d},
-            new Object[] {"TwoCommonMethods", "LORM", 0.26_667d}
-        );
-    }
-
-    @Test
-    public void testsTarget() throws Exception {
+    @ParameterizedTest
+    @MethodSource("targets")
+    public void testsTarget(final String target, final String metric, final double value)
+        throws Exception {
         final Path output = Files.createTempDirectory("");
         new XslReport(
-            new Skeleton(new FakeBase(this.target)).xml(),
-            this.metric, new XslCalculus()
+            new Skeleton(new FakeBase(target)).xml(),
+            metric, new XslCalculus()
         ).save(output);
         final String xpath;
-        if (Double.isNaN(this.value)) {
+        if (Double.isNaN(value)) {
             xpath = "//class[@id='%s' and @value='NaN']";
         } else {
             xpath = "//class[@id='%s' and number(@value)=%.4f]";
         }
         new Assertion<>(
             new FormattedText(
                 "Must exists with target '%s' and value '%s'",
-                this.target, this.value
+                target, value
             ).asString(),
             XhtmlMatchers.xhtml(
                 new TextOf(
-                    output.resolve(String.format("%s.xml", this.metric))
+                    output.resolve(String.format("%s.xml", metric))
                 ).asString()
             ),
             XhtmlMatchers.hasXPaths(
                 String.format(
                     xpath,
-                    this.target, this.value
+                    target, value
                 )
             )
         ).affirm();
     }
+
+    // @checkstyle MethodLengthCheck (200 lines)
+    private static Collection<Arguments> targets() {
+        return new CollectionOf<>(
+            Arguments.of("NoMethods", "NHD", Double.NaN),
```

Comment:
@HDouss This method looks huge and it does load parameters.
So, why not creating a file with a format:
```
Bar, LCOM, 6.0d
Foo, LCOM, 1.0d
.....
```
and use the folllowing:
```
final List<Arguments> lines = new Mapped<>(
            line -> Arguments.of(
                new org.cactoos.iterable.Mapped<>(
                    text -> text.asString(),
                    new Split(
                        line, ","
                    )
                )
            ),
            new Split(
                new TextOf(
                    new ResourceOf("path_to_file")
                ),
                "\n"
            )
        );
```